xc/xenops_server_xen.ml

```diff
@@ -1176,12 +1176,13 @@ module VM = struct
           extras = [];
         } in
       let bridge_of_network = function
-        | Network.Local b -> b
-        | Network.Remote (_, b) -> b in
-      let nics = List.map (fun vif ->
-          vif.Vif.mac,
-          bridge_of_network vif.Vif.backend,
-          vif.Vif.position
+        | Network.Local b -> Some b
+        | Network.Sriov _ -> None
+        | Network.Remote (_, b) -> Some b in
+      let nics = List.filter_map (fun vif ->
+          match bridge_of_network vif.Vif.backend with
+          | Some b -> Some (vif.Vif.mac, b, vif.Vif.position)
+          | None -> None (* TODO: passthrough with QEMU *)
```

Comment:
Hi @robhoes 
Yes. This work has not been determined. And I think I need your help on understanding the logic here:)
1. Following 3 operations will go through the logic `VM_create_device_model`: VM_start, VM_resume, VM_receive_memory; only VM_start is supported for network SR-IOV.
2. It looks like that `VM_create_device_model` will start the QEMU process for the domU, as the data model; for some kind of VGPU, it will start a helper process also.
3. So far I haven't  identified any change on parameters of starting QEMU process for network SR-IOV VF backed vif.

So, ATM, my conclusion is: there is NO change in Device.Dm.start, Device.Dm.restore, etc for network SR-IOV feature. So I could just filter out the network SR-IOV VF backed vif here.

Can I have your comments on this conclusion?  Thanks very much!

Ming

Reply:
That sounds right. Since SR-IOV VIFs aren't actually a VIFs as far as QEMU is concerned, they should be filtered out here.

I would combine the two matches here though, and eliminate `bridge_of_network`.

Reply:
Combined the two matches and removed the `TODO`.
Thanks!